### PR TITLE
TraceService: return results for transactions in block

### LINKED ISSUE #6077: TraceService: Better error handling for internal errors in TransactionProcessor
### Description
The `TraceService` allows us to create plugins with custom tracers. What is missing right now is the ability to programmatically get errors when the `TransactionProcessor` throws an internal error, because of bug in the custom tracers.

### PR DESCRIPTION
**Pull request by daniellehrner**
## PR description

* The TraceService returns the result of the processed transactions
* In case of an internal error during tx processing, the stack trace is added to the error message as well.

## Fixed Issue(s)
fixes #6077

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Deprecations
 
 ### Additions and Improvements
+- TraceService: return results for transactions in block [#6086](https://github.com/hyperledger/besu/pull/6086)
```
Comment (macfarla) [inline CHANGELOG.md:8]:
```suggestion
- TraceService: return results for transactions in block [#6087](https://github.com/hyperledger/besu/pull/6087)
```

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java [hunk 2]
```diff
@@ -500,7 +500,8 @@ public TransactionProcessingResult processTransaction(
       LOG.error("Critical Exception Processing Transaction", re);
       return TransactionProcessingResult.invalid(
           ValidationResult.invalid(
-              TransactionInvalidReason.INTERNAL_ERROR, "Internal Error in Besu - " + re));
+              TransactionInvalidReason.INTERNAL_ERROR,
+              "Internal Error in Besu - " + re + "\n" + printableStackTraceFromThrowable(re)));
```
Comment (macfarla) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java:504]: what does this look like in reality?

Comment (daniellehrner) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java:504]: This is an example:

`Internal Error in Besu - java.lang.IllegalStateException
    at com.google.common.base.Preconditions.checkState(Preconditions.java:486)
    at net.consensys.linea.zktracer.module.hub.stack.Stack.processInstruction(Stack.java:333)
    at net.consensys.linea.zktracer.module.hub.Hub.handleStack(Hub.java:376)
    at net.consensys.linea.zktracer.module.hub.Hub.processStateExec(Hub.java:470)
    at net.consensys.linea.zktracer.module.hub.Hub.tracePreOpcode(Hub.java:705)
    at net.consensys.linea.zktracer.ZkTracer.tracePreExecution(ZkTracer.java:101)
    at org.hyperledger.besu.evm.EVM.runToHalt(EVM.java:187)
    at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.codeExecute(AbstractMessageProcessor.java:174)
    at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.process(AbstractMessageProcessor.java:196)
    at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.process(MainnetTransactionProcessor.java:511)
    at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:418)
    at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:206)
    at org.hyperledger.besu.services.TraceServiceImpl.lambda$trace$6(TraceServiceImpl.java:217)
    at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
    at org.hyperledger.besu.services.TraceServiceImpl.trace(TraceServiceImpl.java:202)
    at org.hyperledger.besu.services.TraceServiceImpl.lambda$trace$4(TraceServiceImpl.java:182)
    at org.hyperledger.besu.ethereum.api.jsonrpc.internal.processor.Tracer.lambda$processTracing$0(Tracer.java:52)
    at org.hyperledger.besu.ethereum.api.query.BlockchainQueries.lambda$getAndMapWorldState$23(BlockchainQueries.java:923)
    at java.base/java.util.Optional.flatMap(Optional.java:289)
    at org.hyperledger.besu.ethereum.api.query.BlockchainQueries.getAndMapWorldState(BlockchainQueries.java:919)
    at `

File: besu/src/main/java/org/hyperledger/besu/services/TraceServiceImpl.java [hunk 3]
```diff
@@ -85,10 +87,41 @@ public void traceBlock(final long blockNumber, final BlockAwareOperationTracer t
...
+  private BlockTraceResult traceBlock(
+      final Optional<Block> maybeBlock, final BlockAwareOperationTracer tracer) {
     checkArgument(tracer != null);
-    final Optional<Block> block = blockchainQueries.getBlockchain().getBlockByHash(hash);
-    block.ifPresent(value -> trace(value, tracer));
+    if (maybeBlock.isEmpty()) {
+      return BlockTraceResult.empty();
+    }
+
+    final Optional<List<TransactionProcessingResult>> results = trace(maybeBlock.get(), tracer);
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/services/TraceServiceImpl.java:101]: is it more efficient to assign a variable to maybeBlock.get() - it's used below as well

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**